float32[] eta
int32 current_waypoint